src/CommandPool.php

```diff
@@ -58,7 +58,7 @@ public function __construct(
                 if ($before) {
                     $before($command, $key);
                 }
-                yield $client->executeAsync($command);
+                yield $key => $client->executeAsync($command);
```

Comment:
Yes, but the `mapRegions` function doesn't make much sense without it. If the iterator provided to the command pool yields meaningful keys, they are only currently passed to the `before` callback, not `fulfilled` or `rejected`.